python/servo/package_commands.py

```diff
@@ -183,9 +182,29 @@ def package(self, release=False, dev=False, android=None, debug=False, debugger=
             tar_path = '/'.join(dir_to_package.split('/')[:-1]) + '/'
             tar_path += datetime.utcnow().replace(microsecond=0).isoformat()
             tar_path += "-servo-tech-demo.tar.gz"
-            with tarfile.open(tar_path, "w:gz") as tar:
-                # arcname is to add by relative rather than absolute path
-                tar.add(dir_to_package, arcname='servo/')
+
+            # Archive deterministically
+            # See https://reproducible-builds.org/docs/archives/
+            archive_root_dir = 'servo'
+            with cd(dir_to_package):
+                p1 = subprocess.Popen(['find', '.', '-print0'],
+                                      stdout=subprocess.PIPE)
+                p2 = subprocess.Popen(['sort', '-z'], env=dict(os.environ, LC_ALL='C'),
+                                      stdin=p1.stdout, stdout=subprocess.PIPE)
+                p3 = subprocess.Popen(['tar', '--null', '-T', '-', '--no-recursion',
+                                       '--mtime', '1970-01-01 00:00Z',
+                                       '--transform', 's,^./,{}/,'.format(archive_root_dir),
+                                       '--owner=root', '--group=root', '--numeric-owner',
+                                       '-cf', '-'],
```

Comment:
Do  the `'-f', '-'` args mean write to stdout? I think the `--to-stdout` option would be more clear.